### ISSUE TITLE
about code

### ISSUE THREAD

**Issue by igodrr** [kwuking/TimeMixer#77]
`class DFT_series_decomp(nn.Module):
    """
    Series decomposition block
    """

    def __init__(self, top_k=5):
        super(DFT_series_decomp, self).__init__()
        self.top_k = top_k

    def forward(self, x):
        xf = torch.fft.rfft(x)
        freq = abs(xf)
        freq[0] = 0
        top_k_freq, top_list = torch.topk(freq, self.top_k)
        xf[freq <= top_k_freq.min()] = 0
        x_season = torch.fft.irfft(xf)
        x_trend = x - x_season
        return x_season, x_trend`
In this code, is the dimension of x (B, L, C), where L is the sequence length and C is the feature dimension? However, I noticed that xf = torch.fft.rfft(x) is performing the FFT along the last dimension (C). Isn’t this somewhat unreasonable? Shouldn’t it be corrected to xf = torch.fft.rfft(x, dim=1)?”  And what does freq[0] = 0 mean in this context? Is this operation being performed on the batch dimension?   Looking forward to your reply, Thanks very much !

**Comment by kwuking** [comment]
Many thanks for your attention and advice.  Your understanding is correct. The FFT operation should be performed along the time dimension. The code here is ambiguous, and we will rectify this issue as soon as possible. Once again, thank you for your insightful suggestion.
